src/xa/xml/maybes.js

```diff
@@ -0,0 +1,159 @@
+import * as _ from 'lodash';
+
+function maybe_find_one(pn, xp, attrs = [], fn = null) {
+  let rv = pn.find(xp)[0];
+
+  if (rv) {
+    attrs = _.reduce(attrs, (o, k) => {
+      const attr = rv.attr(k);
+      return attr ? _.set(o, k, attr.value()) : o
+    }, {})
+  }
+  if (rv && fn) {
+    rv = fn(rv, attrs)
+  }
+
+  return rv;
+}
+
+function maybe_find_many(pn, xp, fn) {
+  let rv = pn.find(xp);
+  if (rv && _.some(rv, (r)=>!!r) && fn) {
+    rv = fn(rv);
+  }
+
+  return rv;
+}
+
+function maybe_find_list(pn, xps, fn) {
```

Comment:
Try to avoid using `_.compact`. I think this will work:

```
let rv = _.reduce(xps, (els, xp) => {
  let n = pn.find(xp);
  return n ? _.concat(els, n) : els
};
```